tldr

```diff
@@ -168,12 +172,23 @@ path_for_cmd() {
     fi
 }
 
+# cache a local copy of the tldrpage
+cache_tldr() {
+    curl -sfL https://github.com/tldr-pages/tldr/tarball/master > $configdir/tldr-pages.tar.gz
+    tar xzf "$configdir"/tldr-pages.tar.gz -C "$configdir"
+    mv -f "$configdir"/tldr-pages-tldr-*/pages/* "$configdir"
```

Comment:
Add --backup to mv so mv doesn't cry that mv: cannot move source to destination: Directory not empty.

Reply:
And what comes to latter, it would be better to do

>for dir in "$configdir"/tldr-pages-tldr-*/pages/*/; do
>	dir=${dir%*/}
>	echo ${dir##*/}
>	mv -f "$dir" "$configdir"
>done

With that there is no copy behavior with --backup nor mv cry Directory not empty.

Reply:
I'm happy to make an update here. Can you help me understand what the problem with the current `mv -f *` behavior is?  `mv -f` should overwrite a directory.

Reply:
Oops, it looks like @ghost is a now-deleted user.  @raylee Do you understand this comment?

Reply:
Working backward on the comments, I'm not sure this:
`dir=${dir%*/}`
does what's intended? The % here matches the shortest regex so the * is superfluous, and the expression will remove only a trailing slash if it exists:
`dir=a/b/c/; echo ${dir%/} ${dir%*/}
a/b/c a/b/c`

Reply:
Ah, I see! I didn't realize `mv -f` behaved that way.  Thank you for the example. 

Perhaps it would be best to do something like this:
```
for dir in "$configdir"/tldr-pages-tldr-*/pages/*; do
    rm -rf "$configdir/$dir"
    mv "$dir" "$configdir"
done
```
That way we don't have to worry about any further nested folders, or about backups already existing.

Reply:
Okay, I've walked through the above code by hand with the current tldr pages and see what the OP is trying to do now. But I think there's an easier way.

Both GNU and BSD tar support a `--strip-components=n` option, which removes the first `n` path components from the filenames during extraction. So the above can be simplified to:

```
curl -sfL https://github.com/tldr-pages/tldr/tarball/master > "$configdir"/tldr-pages.tar.gz
tar xzf "$configdir"/tldr-pages.tar.gz --strip-components=2 -C "$configdir"
```
the `mv` can then be removed, and there'll be no need for the for loop suggestion above.

Sound good?

Reply:
Great solution - thanks!